idaes/models_extra/temperature_swing_adsorption/fixed_bed_tsa0d.py

```diff
@@ -0,0 +1,4522 @@
+        ] = self.flue_gas_processed_year
+
+        var_dict[
+            "Amount of flue gas processed per year (target) [Gmol/year]"
+        ] = self.flue_gas_processed_year_target
+
+        var_dict["Amount of CO2 to atmosphere [mol/s]"] = self.emissions_co2
+
+        var_dict[
+            "Concentration of CO2 emitted to atmosphere [ppm]"
+        ] = self.emissions_co2_ppm
+
+        return var_dict
+
+    def summary(self, export=False):
+
+        var_dict = self._var_dict()
+
+        summary_dir = {}
+        summary_dir["Value"] = {}
+        summary_dir["pos"] = {}
+
+        count = 1
+        for k, v in var_dict.items():
+            summary_dir["Value"][k] = value(v)
+            summary_dir["pos"][k] = count
+            count += 1
+
+        df = DataFrame.from_dict(summary_dir, orient="columns")
+        del df["pos"]
+        if export:
+            df.to_csv(f"{self.local_name}_summary.csv")
+
+        print("\n" + "="*84)
+        print(f"summary {self.local_name}")
+        print("-"*84)
+        stdout.write(
+            textwrap.indent(
+                stream_table_dataframe_to_string(df),
+                " "*4))
+        print("\n" + "="*84 + "\n")
+
+    def _get_performance_contents(self, time_point=0):
+
+        var_dict = self._var_dict()
+
+        rem_from_var_dict = []
+        for key, val in var_dict.items():
+            if isinstance(val, Var):
+                pass
+            else:
+                rem_from_var_dict.append(key)
+
+        for r in rem_from_var_dict:
+            del var_dict[r]
+
+        return {"vars": var_dict}
+
+    def _get_stream_table_contents(self, time_point=0):
+
+        stream_attributes = {}
+        stream_attributes["Units"] = {}
+
+        for n, v in {"Inlet": "inlet",
+                     "CO2 Rich Stream": "co2_rich_stream",
+                     "N2 Rich Stream": "n2_rich_stream",
+                     "H20 Stream": "h2o_o2_stream"}.items():
+            port_obj = getattr(self, v)
+
+            stream_attributes[n] = {}
+
+            for k in port_obj.vars:
+                for i in port_obj.vars[k].keys():
+                    if isinstance(i, float):
+                        var = port_obj.vars[k][time_point]
+                        stream_attributes[n][k] = value(var)
+                        stream_attributes["Units"][k] = units.get_units(var)
+                    else:
+                        if len(i) == 2:
+                            kname = str(i[1])
+                        else:
+                            kname = str(i[1:])
+                        var = port_obj.vars[k][time_point, i[1:]]
+                        stream_attributes[n][k+" "+kname] = value(var)
+                        stream_attributes["Units"][k+" "+kname] = \
+                            units.get_units(var)
+
+        return DataFrame.from_dict(stream_attributes, orient="columns")
+
+    def _get_costing_electric_boiler(self):
```

Comment:
Should this be part of the unit model file? Costing is generally something separate from the unit model. This would also help cut down the size of this file (which is huge).

Reply:
moved costing to separate file